### PR TITLE
[KYUUBI #3805] Skip PySpark tests for nightly builds

### LINKED ISSUE #3805: [GA] `Kyuubi Spark Nightly` failed with PySpark-related tests.
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What kind of test improve?

- [ ] Fix flaky tests.
- [ ] Fix bug in tests.
- [ ] Increase test coverage.
- [X] Other kinds of test improve.

### Describe the issues of the existing tests or improvements for new tests

https://github.com/apache/incubator-kyuubi/actions/runs/3458458256/jobs/5772898116

```
- pyspark support *** FAILED ***
  org.apache.kyuubi.jdbc.hive.KyuubiSQLException: SPARK_HOME not found!
  at org.apache.kyuubi.jdbc.hive.Utils.verifySuccess(Utils.java:85)
  at org.apache.kyuubi.jdbc.hive.Utils.verifySuccessWithInfo(Utils.java:71)
  at org.apache.kyuubi.jdbc.hive.KyuubiStatement.runAsyncOnServer(KyuubiStatement.java:272)
  at org.apache.kyuubi.jdbc.hive.KyuubiStatement.executeWithConfOverlay(KyuubiStatement.java:195)
  at org.apache.kyuubi.jdbc.hive.KyuubiStatement.execute(KyuubiStatement.java:190)
  at org.apache.kyuubi.jdbc.hive.KyuubiStatement.executeQuery(KyuubiStatement.java:397)
  at org.apache.kyuubi.engine.spark.operation.PySparkTests.$anonfun$runPySparkTest$1(PySparkTests.scala:66)
  at org.apache.kyuubi.engine.spark.operation.PySparkTests.$anonfun$runPySparkTest$1$adapted(PySparkTests.scala:64)
  at org.apache.kyuubi.operation.JDBCTestHelper.$anonfun$withMultipleConnectionJdbcStatement$3(JDBCTestHelper.scala:60)
  at org.apache.kyuubi.operation.JDBCTestHelper.$anonfun$withMultipleConnectionJdbcStatement$3$adapted(JDBCTestHelper.scala:60)
  ...
  Cause: java.lang.IllegalStateException: SPARK_HOME not found!
  at org.apache.kyuubi.engine.spark.operation.ExecutePython$.$anonfun$defaultSparkHome$6(ExecutePython.scala:153)
  at scala.Option.getOrElse(Option.scala:189)
  at org.apache.kyuubi.engine.spark.operation.ExecutePython$.defaultSparkHome(ExecutePython.scala:153)
  at org.apache.kyuubi.engine.spark.operation.ExecutePython$.$anonfun$createSessionPythonWorker$2(ExecutePython.scala:128)
  at scala.collection.immutable.HashMap$HashTrieMap.getOrElse0(HashMap.scala:596)
  at scala.collection.immutable.HashMap$HashTrieMap.getOrElse0(HashMap.scala:594)
  at scala.collection.immutable.HashMap.getOrElse(HashMap.scala:73)
  at org.apache.kyuubi.engine.spark.operation.ExecutePython$.createSessionPythonWorker(ExecutePython.scala:128)
  at org.apache.kyuubi.engine.spark.operation.SparkSQLOperationManager.$anonfun$newExecuteStatementOperation$7(SparkSQLOperationManager.scala:96)
  at scala.collection.concurrent.Map.getOrElseUpdate(Map.scala:97)
  ...
```

we can disable the PySpark-related tests for `Kyuubi Spark Nightly`

### Are you willing to submit PR?

- [ ] Yes. I can submit a PR independently to improve.
- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

to close #3805 .

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -30,6 +30,8 @@ jobs:
       matrix:
         profiles:
           - '-Pspark-master -pl externals/kyuubi-spark-sql-engine -am'
+        exclude-tags:
+          - '-Dmaven.plugin.scalatest.exclude.tags=org.scalatest.tags.PySparkTest'
```
Comment (pan3793) [inline .github/workflows/nightly.yml:34]: add to pom.xml instead

Comment (bowenliang123) [reply .github/workflows/nightly.yml]: ok, moved to pom.xml and remove this `exclude-tags` from nightly workflow.

**Comment by bowenliang123** [comment]
cc @cfmcgrady

**Review by pan3793 — APPROVED** [review]

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Comment by cfmcgrady** [comment]
later LGTM.

**Comment by cfmcgrady** [comment]
Actually, the annotation not works for scala `trait` :(

**Comment by cfmcgrady** [comment]
the nightly builds fail again

FYI: https://github.com/apache/incubator-kyuubi/actions/runs/3529101434/jobs/5919834956

**Comment by bowenliang123** [comment]
Sorry to hear that :(
Or let me move the annotation to every test case in trait `PySparkTests`.
What do you think?  @cfmcgrady

**Pull request closed without merging**